projectile.el

```diff
@@ -1181,6 +1181,9 @@ Files are returned as relative paths to DIRECTORY."
 The function calls itself recursively until all sub-directories
 have been indexed.  The PROGRESS-REPORTER is updated while the
 function is executing."
+  (projectile-index-directory--internal directory patterns progress-reporter (projectile-ignored-files) (projectile-ignored-directories)))
+
+(defun projectile-index-directory--internal (directory patterns progress-reporter ignored-files ignored-directories)
```

Comment:
I think it'd be simpler to just let-bind the ignored files/folders and not introduce new helper functions.